File: src/main/java/ru/timmson/kanban/game/model/Stage.java
package ru.timmson.kanban.game.model;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

@Getter
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@ToString
public class Stage {

    public static Stage STAGE_BACKLOG = new Stage("B");

    public static Stage STAGE_ANALYSIS = new Stage("A");

    public static Stage STAGE_DEVELOPMENT = new Stage("D");

    public static Stage STAGE_TESTING = new Stage("T");

    public static Stage STAGE_RELEASE = new Stage("R");

    private String type;

}

File: src/main/kotlin/ru/timmson/kanban/game/model/Stage.kt
package ru.timmson.kanban.game.model

data class Stage constructor(var stage: String) {

    companion object {
        @JvmField
        val STAGE_BACKLOG = Stage("backlog")

        @JvmField
        val STAGE_ANALYSIS = Stage("analysis")

        @JvmField
        val STAGE_DEVELOPMENT = Stage("development")

        @JvmField
        val STAGE_TESTING = Stage("testing")

        @JvmField
        val STAGE_RELEASE = Stage("release")
    }

    override fun toString(): String = stage

}